batch7crud-roll3/src/main/java/com/lftechnology/batch7crud/dao/StudentDAO.java

```diff
@@ -0,0 +1,125 @@
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public List<Student> fetch(int page, int pageSize) throws DataException {
+    List<Student> studentList = new ArrayList<Student>();
+
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_LIST);) {
+      stmnt.setInt(1, pageSize);
+      stmnt.setInt(2, (page - 1) * pageSize);
+
+      try (ResultSet result = stmnt.executeQuery();) {
+        while (result.next()) {
+          Student student = new Student();
+          student.setId(result.getInt("id"));
+          student.setRoll(result.getInt("roll"));
+          student.setName(result.getString("name"));
+          studentList.add(student);
+        }
+      }
+
+      return studentList;
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public int fetchTotal() throws DataException {
+    int totalSize = 0;
+
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_FETCH_TOTAL);
+        ResultSet result = stmnt.executeQuery();) {
+      while (result.next()) {
+        totalSize = result.getInt("total");
+      }
+      return totalSize;
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Student fetchStudentById(int id) throws DataException {
+    Student student = null;
+
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_FETCH_BY_ID);) {
+      stmnt.setInt(1, id);
+
+      try (ResultSet result = stmnt.executeQuery();) {
+
+        while (result.next()) {
+          student = new Student();
+          student.setId(result.getInt("id"));
+          student.setRoll(result.getInt("roll"));
+          student.setName(result.getString("name"));
+        }
+        return student;
+      }
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void edit(Student student, int id) throws DataException {
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_EDIT);) {
+      stmnt.setInt(1, student.getRoll());
+      stmnt.setString(2, student.getName());
+      stmnt.setInt(3, id);
+      stmnt.executeUpdate();
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void delete(int id) throws DataException {
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_DELETE);) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)